compendium/DeclarativeServices/src/metadata/MetadataParserImpl.cpp

```diff
@@ -38,6 +38,12 @@ namespace cppmicroservices
         namespace metadata
         {
 
+            std::string
+            removeLeadingNamespacing(std::string const& className)
+            {
+                return className.substr(className.find_first_not_of(':'));
```

Comment:
I don't think calling `std::string::substr` with `std::string::npos` (if there aren't any leading colons) is defined behaviour? It could potentially throw?

Reply:
I am not sure I understand what situation you think is dangerous, but it is `find_first_not_of` so it isn't an issue if there are no leading `::` as it will just resolve to the 0th letter. 

What is dangerous, however, is if the classname is `""` or `":::::"`, in which case this will throw. So I need to fix those edge cases. 

Can you clarify if there is another situation that you see as potentially problematic?

Reply:
Sorry, yep, right you are! So handle the npos before calling substr and return the string as it was passed in?

Reply:
Indeed!